notes/privacy.md

```diff
@@ -0,0 +1,257 @@
+# DialogDB Privacy RFC
+
+## Design Goals
+
+DialogDB's privacy model is designed with the following goals:
+
+1. **Full Infrastructure Privacy**: Storage and coordination infrastructure should have no access to user data
+2. **Tiered Access Levels**: Different actors can be granted different levels of access
+3. **Selective Sharing**: Fine-grained control over which parts of the database are shared
+4. **Authorization Integration**: Support for capability-based authorization systems like UCAN
+5. **Privacy-Efficiency Tradeoffs**: Allow users to choose their preferred balance
+
+## Access Levels
+
+DialogDB implements a layered privacy model with distinct access levels:
+
+```mermaid
+flowchart TD
+    subgraph "Access Levels"
+        direction TB
+        L0["Level 0: No Access<br/>Complete Opacity"]
+        L1["Level 1: Structure Access<br/>Blob Connectivity"]
+        L2["Level 2: Range Access<br/>Tree Navigation"]
+        L3["Level 3: Data Access<br/>Content Visibility"]
+    end
+    
+    L0 --> L1 --> L2 --> L3
+    
+    style L0 fill:#ddd,stroke:#333,stroke-width:1px
+    style L1 fill:#bfb,stroke:#333,stroke-width:1px
+    style L2 fill:#bbf,stroke:#333,stroke-width:1px
+    style L3 fill:#f9f,stroke:#333,stroke-width:1px
```

Comment:
Nit: color choices here make the text difficult to read

![image](https://github.com/user-attachments/assets/cab93e1e-181a-46d6-aaed-951e058019ce)